File: Barone.Client/src/app/modules/shared/models/insumo.model.ts
import { Constructable } from "src/app/core/models/constructable";

export class InsumoModel extends Constructable<Partial<InsumoModel>> {
    public id: number;
    public Nombre: string;
    public TipoUnidadMedida: number;
    public Stock: number;

    public static fromJSON(json: any): InsumoModel {
        let insumo = Object.create(InsumoModel.prototype);
        return Object.assign(insumo, json, {
            created: new Date(json.created)
        });
    }
}

File: Barone.api/Models/InsumoModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Barone.api.Models
{
    public class InsumoModel
    {
        public int id { get; set; }
        public string Nombre { get; set; }
        public int TipoUnidadMedida { get; set; }
        public int Stock { get; set; }


    }
}